system/inc/system_sleep_configuration.h

```diff
@@ -253,6 +256,27 @@ class SystemSleepConfiguration: protected SystemSleepConfigurationHelper {
         return *this;
     }
 
+    SystemSleepConfiguration& gpios(const Vector<std::pair<pin_t, InterruptMode>>& pins) {
```

Comment:
In case of an existing entry do you just update the interrupt mode?  Take a look at https://github.com/particle-iot/tracker-edge/blob/develop/src/tracker_sleep.cpp#L115-L126 for reference

Reply:
The `gpio(pin_t pin, InterruptMode mode)` overload will take care of that: https://github.com/particle-iot/device-os/blob/feature/sleep_pin_list/ch60648/system/inc/system_sleep_configuration.h#L236

Reply:
Great!